lib/contact_screen.dart

```diff
@@ -0,0 +1,317 @@
+import 'package:fast_contacts/fast_contacts.dart';
+import 'package:flutter/material.dart';
+import 'package:kilvish/models.dart';
+import 'package:kilvish/provider/search_provider.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:collection/collection.dart';
+
+enum ContactSelection { singleSelect, multiSelect }
+
+class ContactScreen extends StatefulWidget {
+  final ContactSelection contactSelection;
+
+  const ContactScreen(
+      {Key? key, this.contactSelection = ContactSelection.singleSelect})
+      : super(key: key);
+
+  @override
+  State<ContactScreen> createState() => _ContactScreenState();
+}
+
+class _ContactScreenState extends State<ContactScreen> {
+  List<ContactModel> _selectedContactsList = [];
+  List<ContactModel> _contactsList = [
+    ContactModel(
+        kilvishId: "Kelvish ID 1",
+        name: 'Kilvish User 1',
+        phoneNumber: "65656-52452"),
+  ];
+  bool _isLoading = true;
+
+  final SearchNotifier _searchNotifier = SearchNotifier();
+  final ValueNotifier<bool> _valueNotifier = ValueNotifier<bool>(true);
+  final TextEditingController _searchController = TextEditingController();
+  String _filterOn = "name";
+  final String _hintText = "Enter name";
+  final String _appbarTitle = "Contact List";
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+      await getContactPermission();
+    });
+  }
+
+
+  @override
+  void dispose() {
+    _searchController.dispose();
+    super.dispose();
+  }
+
+  Future<void> getContactPermission() async {
+    if (await Permission.contacts.request().isGranted) {
+      // Permission is granted, fetch contacts
+      fetchContacts();
+    } else {
+      final status = await Permission.contacts.request();
+      if (status.isGranted) {
+        // Permission is granted, fetch contacts
+        fetchContacts();
+      }
+    }
+  }
+
+  Future fetchContacts() async {
+    List<Contact> contacts = await FastContacts.getAllContacts(batchSize: 5);
+    if (contacts.length > 5) {
+      contacts = contacts.take(5).toList();
+    }
```

Comment:
I think the above logic has a flaw when it will come to filtering. We need to fetch all contacts & then keep them as state variable (something like `_allContacts`) .. when filtering happens, we need to iterate on all contacts & show only top 5 that matches the searchText. 

I was actually hoping that we use a library that takes the search query & then return us only 5-10 such results. But I noted that getAllContacts() from FastContacts does not take search query. I think it is not the best suited library for us. I am still ok using it but then the implementation logic has to change as mentioned in the previous paragraph.